### ISSUE TITLE
childPages vs parent

### ISSUE THREAD

**Issue by TylerBarnes** [wp-graphql/wp-graphql#1078]
I'm noticing that there's a discrepancy between how child and parent nodes are handled. For parents we have a PostObjectUnion type but children are the same type as the current node. Seems like they should work in the same way, though I may be missing something! The naming is also different `childPages` vs `parent`, should it also be `parentPage`?

```gql
{
  pages {
    nodes {
      title
      childPages {
        nodes {
          title
          parent {
            __typename
            ... on Page {
              title
            }
          }
        }
      }
    }
  }
}
```

```json
{
  "data": {
    "pages": {
      "nodes": [
        {
          "title": "Sample Page",
          "childPages": {
            "nodes": [
              {
                "title": "Child page",
                "parent": {
                  "__typename": "Page",
                  "title": "Sample Page"
                }
              }
            ]
          }
        }
      ]
    }
  }
}
```

**Comment by jasonbahl** [comment]
@TylerBarnes in WordPress, a parent of a Post can be of any _other_ Post Type, it doesn't have to be of the same Post Type, which is why it returns a `Union`. This will _probably_ change to return an Interface though, now that I have the `ContentNode` Interface in the works. 

The naming is `parent` instead of `parentPage` as the parent might not be a page. 

The `childPages` connection specifically returns children of the same Type. Children _can_ be of any Post type though, so perhaps we _should_ also change this to be `children` and be a connection that returns a `ContentNode` interface as children _can_ be of any Type, not only the same type as the parent.

**Comment by TylerBarnes** [comment]
Oh, interesting, I didn't know you can have those relationships between different post types, that's cool! It might be good to change to `children` then like you mentioned.

**Comment by jasonbahl** [comment]
@TylerBarnes this is addressed in #1337 (...what isn't #AmIRight 😂)

Anyway, here's an example of this in action:

**Comment by ltroya-as** [comment]
Is it possible to have this connection between posts and custom post types? @jasonbahl. 

I am using the download monitor plugin and it handles two CPT: `dlm_download` and `dlm_download_version`

- `dlm_download` is the CPT
- `dlm_download_version` is a child CPT where the file info is managed
